frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs

```diff
@@ -71,271 +97,322 @@
        {:class (dom/classnames :selected (= padding-type :simple))
-        :alt (tr "workspace.options.layout.padding-simple")
+        :alt "Padding"
         :on-click #(on-change-style :simple)}
        i/auto-padding]
       [:div.padding-icon.tooltip.tooltip-bottom
        {:class (dom/classnames :selected (= padding-type :multiple))
-        :alt (tr "workspace.options.layout.padding")
+        :alt "Padding - sides"
         :on-click #(on-change-style :multiple)}
        i/auto-padding-side]]
-
-     (cond
-       (= padding-type :simple)
-       [:div.tooltip.tooltip-bottom
-        {:alt (tr "workspace.options.layout.padding-all")}
-        [:div.input-element.mini
-
-         [:> numeric-input
-          {:placeholder "--"
-           :on-click #(dom/select-target %)
-           :on-change (partial on-change :simple)
-           :value (:p1 (:layout-padding values))}]]]
-
-       (= padding-type :multiple)
-       (for [num [:p1 :p2 :p3 :p4]]
-         [:div.tooltip.tooltip-bottom
-          {:key (dm/str "padding-" (d/name num))
-           :alt (case num
-                  :p1 (tr "workspace.options.layout.top")
-                  :p2 (tr "workspace.options.layout.right")
-                  :p3 (tr "workspace.options.layout.bottom")
-                  :p4 (tr "workspace.options.layout.left"))}
-          [:div.input-element.mini
-           [:> numeric-input
-            {:placeholder "--"
-             :on-click #(dom/select-target %)
-             :on-change (partial on-change num)
-             :value (num (:layout-padding values))}]]]))]))
+     [:div.wrapper
+      (cond
+        (= padding-type :simple)
+        [:div.tooltip.tooltip-bottom
+         {:alt (tr "workspace.options.layout.padding-all")}
+         [:div.input-element.mini
+
+          [:> numeric-input
+           {:placeholder "--"
+            :on-click #(dom/select-target %)
+            :on-change (partial on-change :simple)
+            :value rx}]]]
+
+        (= padding-type :multiple)
+        (for [num [:p1 :p2 :p3 :p4]]
+          [:div.tooltip.tooltip-bottom
+           {:key (dm/str "padding-" (d/name num))
+            :alt (case num
+                   :p1 "Top"
+                   :p2 "Right"
+                   :p3 "Bottom"
+                   :p4 "Left")}
+           [:div.input-element.mini
+            [:> numeric-input
+             {:placeholder "--"
+              :on-click #(dom/select-target %)
+              :on-change (partial on-change num)
+              :value (num (:layout-padding values))}]]]))]]))
+
+(mf/defc gap-section
+  [{:keys [gap-selected? set-gap gap-value toggle-gap-type gap-type]}]
+  (let [gap-locked?  (= gap-type :simple)]
+    [:div.gap-group
+     [:div.gap-row.tooltip.tooltip-bottom-left
+      {:alt "Row gap"}
+      [:span.icon
+       {:class (dom/classnames :activated (or (and (not gap-locked?) (= @gap-selected? :row-gap)) (and gap-locked? (not= @gap-selected? :none))))}
```

Comment:
I think it's better to extract this complex condition to the let so it's easier to read.

Reply:
you are right